dbms/src/Interpreters/DDLWorker.cpp

```diff
@@ -622,14 +622,14 @@ void DDLWorker::processTask(DDLTask & task, const ZooKeeperPtr & zookeeper)
 
             if (auto query_with_table = dynamic_cast<ASTQueryWithTableAndOutput *>(rewritten_ast.get()); query_with_table)
             {
-                String database = query_with_table->database.empty() ? context.getCurrentDatabase() : query_with_table->database;
-                StoragePtr storage = context.tryGetTable(database, query_with_table->table);
+                String database = query_with_table->databaseName(context.getCurrentDatabase());
+                StoragePtr storage = context.tryGetTable(database, query_with_table->tableName());
```

Comment:
If it's `CREATE/DROP DATABASE` query, `tableName()` will throw an exception (you can see `Code: 223, e.displayText() = DB::Exception: AST node is nullptr` in failed tests). Probably there are some other places in code, where a table name is expected to be empty string. It's better to find and rewrite all such places, but returning empty string from `tableName()` when `table` node is `nullptr` will also be ok